Running in Docker using container name audubon_racing_id10_container and image name audubon_racing_image.
Sending build context to Docker daemon  31.06MB
Step 1/13 : FROM ros:noetic-robot-focal
 ---> 8b211266230d
Step 2/13 : RUN echo "Starting..."
 ---> Using cache
 ---> 132d939a8cce
Step 3/13 : RUN apt-get update && apt-get install -y python3-pip ros-noetic-ackermann-msgs
 ---> Using cache
 ---> 192880db760e
Step 4/13 : ARG TARFILE="dockerfile-needs-arg.tar.gz"
 ---> Using cache
 ---> a9bce34ba3cc
Step 5/13 : RUN echo "Starting with tarfile $TARFILE..."
 ---> Using cache
 ---> b77eab717fea
Step 6/13 : WORKDIR /race
 ---> Using cache
 ---> 42891b5f0c43
Step 7/13 : ADD "./${TARFILE}" /race/files
 ---> Using cache
 ---> 13fa9bc2bdea
Step 8/13 : RUN mkdir /catkin_ws && mkdir /catkin_ws/src && cp -r /race/files/pkg /catkin_ws/src/pkg
 ---> Using cache
 ---> 92c560393f25
Step 9/13 : COPY ./networking.py .
 ---> Using cache
 ---> 992a3b93e568
Step 10/13 : COPY ./riders_client_tcp_race.py .
 ---> 9a273e024bc6
Step 11/13 : RUN pip3 install numpy==1.17.4 pyyaml==5.3.1
 ---> Running in 7548f28a6e5e
Requirement already satisfied: numpy==1.17.4 in /usr/lib/python3/dist-packages (1.17.4)
Requirement already satisfied: pyyaml==5.3.1 in /usr/lib/python3/dist-packages (5.3.1)
Removing intermediate container 7548f28a6e5e
 ---> dd074e166ba8
Step 12/13 : WORKDIR /race/files/pkg
 ---> Running in 42eed09ddae9
Removing intermediate container 42eed09ddae9
 ---> c25fdc6c3199
Step 13/13 : RUN pip3 install -r requirements.txt
 ---> Running in 231b0195135f
Collecting Pillow==8.3.2
  Downloading Pillow-8.3.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.0 MB)
Installing collected packages: Pillow
Successfully installed Pillow-8.3.2
Removing intermediate container 231b0195135f
 ---> bf4109b7eae3
Successfully built bf4109b7eae3
Successfully tagged audubon_racing_image:latest
running Driver import line: from pkg.vfh_gym import VFH as Driver
Connecting to port 38163...
Traceback (most recent call last):
  File "/race/riders_client_tcp_race.py", line 92, in start_client
    speed, steer = driver.process_observation(ranges=scan, ego_odom=odom)
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 270, in process_observation
    self.pad_sectors()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 338, in pad_sectors
    alpha = sec.get_theta_pad_radius()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 158, in get_theta_pad_radius
    return math.asin(R_car/self.max_dis_to_go)
ValueError: math domain error
Traceback (most recent call last):
  File "/race/riders_client_tcp_race.py", line 92, in start_client
    speed, steer = driver.process_observation(ranges=scan, ego_odom=odom)
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 270, in process_observation
    self.pad_sectors()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 338, in pad_sectors
    alpha = sec.get_theta_pad_radius()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 158, in get_theta_pad_radius
    return math.asin(R_car/self.max_dis_to_go)
ValueError: math domain error
Traceback (most recent call last):
  File "/race/riders_client_tcp_race.py", line 92, in start_client
    speed, steer = driver.process_observation(ranges=scan, ego_odom=odom)
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 270, in process_observation
    self.pad_sectors()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 338, in pad_sectors
    alpha = sec.get_theta_pad_radius()
  File "/race/files/pkg/src/pkg/vfh_gym.py", line 158, in get_theta_pad_radius
    return math.asin(R_car/self.max_dis_to_go)
ValueError: math domain error
Error running driver code: math domain error
Error running driver code: math domain error
Error running driver code: math domain error
got 'exit' or None: {'type': 'exit'}
